### PR TITLE
chore: remove unsed vars & cleanups

### PR DESCRIPTION
**Pull request by MatissJanis**
Removing a couple more unused variables + a tiny improvement in the compiler to make some more tests pass.

### CODE REVIEW

File: packages/loot-core/src/server/aql/compiler.js [hunk 1]
```diff
@@ -560,27 +560,24 @@ const compileFunction = saveStack('function', (state, func) => {
...
       let [arg1] = valArray(state, args, ['string']);
       return typed(`LOWER(${arg1})`, 'string');
     }
 
     // integer/float functions
     case '$neg': {
       validateArgLength(args, 1);
-      // eslint-disable-next-line no-unused-vars
-      let [arg1] = valArray(state, args, ['float']);
+      valArray(state, args, ['float']);
```
Comment (MatissJanis) [inline packages/loot-core/src/server/aql/compiler.js:570]: Throws an exception if the vars are invalid, so we can't remove it.

File: packages/loot-core/src/server/spreadsheet/spreadsheet.test.js [hunk 2]
```diff
@@ -109,40 +107,40 @@ describe('Spreadsheet', () => {
...
-  //   spreadsheet.endTransaction();
-
-  //   return new Promise(resolve => {
-  //     spreadsheet.onFinish(() => {
-  //       expect(spreadsheet.getValue('g!foo')).toMatchSnapshot();
-  //       resolve();
-  //     });
-  //   });
-  // });
+  test('querying transactions works', async () => {
```
Comment (MatissJanis) [inline packages/loot-core/src/server/spreadsheet/spreadsheet.test.js:110]: Only uncommenting these tests. No changes in them.

File: packages/loot-core/src/server/spreadsheet/spreadsheet.js [hunk 3]
```diff
@@ -183,7 +183,7 @@ export default class Spreadsheet {
       this.events.emit('change', { names: this.computeQueue });
 
       // Cache the updated cells
-      if (this.saveCache) {
+      if (typeof this.saveCache === 'function') {
```
Comment (MatissJanis) [inline packages/loot-core/src/server/spreadsheet/spreadsheet.js:186]: Making the tests pass.

**Comment by TechwizEE** [comment]
Performed the following steps and verified the test cases pass
- Separately cloned `master` and `matiss/unused-vars`
- Ran command `yarn test:node spreadsheet.test.js` in folder `actual/packages/loot-core`
- Saw the two un-commented tests were passing
- **Total passing testcases: 4 of 4**

Performed the following steps to verify the code change to `spreadsheet.js` line 186
- In file `spreadsheet.js`, reverted line 186 to `master` (i.e. `if (typeof this.saveCache === 'function') {` to `if (this.saveCache) {`)
- Re-ran test, saw test case `querying deep join works` fail with `TypeError: this.saveCache is not a function`
- Reverted to `matiss/unused-vars`, re-ran test again, saw `querying deep join works` pass

Please let me know if there are additional steps/actions I can take to assist in resolving this issue.

**Comment by trevdor** [comment]
Thanks for reviewing, @TechwizEE!

**Pull request merged**